src/java/com/phenix/pct/ServerProcess.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2005-2017 Riverside Software
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package com.phenix.pct;
+
+import java.io.File;
+import java.util.List;
+
+import org.apache.tools.ant.BuildException;
+
+public class ServerProcess extends PCTRun {
+    private String activateProc = null;
+    private String deactivateProc = null;
+    private String connectProc = null;
+    private String disconnectProc = null;
+    private String startupProc = null;
+    private String shutdownProc = null;
+
+    public ServerProcess() {
+        super(false, false);
+    }
+
+    public void setActivateProc(String activateProc) {
+        this.activateProc = activateProc;
+    }
+
+    public void setConnectProc(String connectProc) {
+        this.connectProc = connectProc;
+    }
+
+    public void setDeactivateProc(String deactivateProc) {
+        this.deactivateProc = deactivateProc;
+    }
+
+    public void setDisconnectProc(String disconnectProc) {
+        this.disconnectProc = disconnectProc;
+    }
+
+    public void setShutdownProc(String shutdownProc) {
+        this.shutdownProc = shutdownProc;
+    }
+
+    public void setStartupProc(String startupProc) {
+        this.startupProc = startupProc;
+    }
+
+    @Override
+    public void setProcedure(String procedure) {
+        throw new BuildException("Unknown attribute");
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Unknown attribute" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1192)